utils/misc.ml

```diff
@@ -530,24 +530,39 @@ module Stdlib = struct
       val ignore_m : 'a t -> unit t
       val all : 'a t list -> 'a list t
       val all_unit : unit t list -> unit t
+
+      module Syntax : sig
+        val (let+) : 'a t -> ('a -> 'b) -> 'b t
+        val (and+) : 'a t -> 'b t -> ('a * 'b) t
+        val (let*) : 'a t -> ('a -> 'b t) -> 'b t
+        val (and*) : 'a t -> 'b t -> ('a * 'b) t
+      end
     end
 
     module type S2 = sig
       type ('a, 'e) t
 
       val bind : ('a, 'e) t -> ('a -> ('b, 'e) t) -> ('b, 'e) t
+      val (>>=) : ('a, 'e) t -> ('a -> ('b, 'e) t) -> ('b, 'e) t
       val return : 'a -> ('a, _) t
       val map : ('a -> 'b) -> ('a, 'e) t -> ('b, 'e) t
       val join : (('a, 'e) t, 'e) t -> ('a, 'e) t
       val ignore_m : (_, 'e) t -> (unit, 'e) t
       val all : ('a, 'e) t list -> ('a list, 'e) t
       val all_unit : (unit, 'e) t list -> (unit, 'e) t
+
+      module Syntax : sig
+        val (let+) : ('a, 'e) t -> ('a -> 'b) -> ('b, 'e) t
+        val (and+) : ('a, 'e) t -> ('b, 'e) t -> ('a * 'b, 'e) t
+        val (let*) : ('a, 'e) t -> ('a -> ('b, 'e) t) -> ('b, 'e) t
+        val (and*) : ('a, 'e) t -> ('b, 'e) t -> ('a * 'b, 'e) t
+      end
     end
 
-    module Make (X : Basic) = struct
+    module[@inline] Make (X : Basic) = struct
```

Comment:
It seems worthwhile to make `Make` and `Make2` share an implementation, a la `Base.Monad`.

Reply:
And I think the same can be true of the signatures:
```
module type S = sig
  type 'a t
  include S2 with type ('a, 'e) t := 'a t
end
```

Reply:
Good ideas.